### DISCUSSION TITLE
Zim and Python 3.11

### DISCUSSION THREAD

**Discussion by ctreleaven** [zim-desktop-wiki/zim-desktop-wiki#2282] [General]
Hi:

I contribute a bit to the MacPorts project.  I'm not a Zim user but I noticed that Zim on MacPorts needs to be updated to 0.75.1.  The existing version uses Python 3.7*.  If I update our 'port', is there any reason not to use Python 3.11?  Any known problems?

Craig
*We package applications such as Zim to use a particular version of Python.  Not just whatever is available on a user's system as it makes tracking down bugs much easier.

**Comment by dehesselle** [comment]
While I'm not aware of any problems with 3.11 - but I'm also not using 3.11 anywhere - the official Zim.app ships with Python 3.10 because that's the version used in CI to run the tests. (Tests on Linux are run against 3.6 and 3.10)

**Comment by ctreleaven** [comment]
Thanks for the quick response.  I'll keep it at 3.10 for the time being.  It is easy enough to update to 3.11 at any point.

Also, thanks for reminding me about the test suite.  When I ran it with Python 3.10, I got 3 errors.  I've pasted the text of test_results.html to:

https://paste.macports.org/d9f36aa3ddfb

Should I be concerned about these?

Craig

**Comment by ctreleaven** [comment]
Just wanted to confirm that I've updated Zim on MacPorts.  More info at:

https://ports.macports.org/port/zim/details/

BTW, our buildbot fleet was busy rebuilding gcc and cousins just when I committed the Zim update.  For some reason, we fell out of the queue.  There is an update to gtk3 coming that ought to fix problems on Ventura systems.  I'll nudge Zim after that.

Craig

**Reply by dehesselle** [reply]
Please tell me about your experience with Ventura. I've tested the latest GTK 3.24.36 yesterday and Zim is still unusable for me. See https://github.com/zim-desktop-wiki/zim-desktop-wiki/issues/2219

**Comment by jaap-karssenberg** [comment]
The automated testing on GitHub now covers python 3.11 but for Linux. So
python version in principle supported.

Regards,

Jaap

Op vr 6 jan. 2023 16:21 schreef René de Hesselle ***@***.***>:
